numba/runtime/_nrt_python.c

```diff
@@ -0,0 +1,539 @@
+    return PyLong_FromVoidPtr(mi);
+}
+
+/*
+ * Create a new MemInfo with a new NRT allocation
+ */
+static
+PyObject*
+meminfo_alloc(PyObject *self, PyObject *args) {
+    MemInfo *mi;
+    Py_ssize_t size;
+    if (!PyArg_ParseTuple(args, "K", &size)) {
+        return NULL;
+    }
+    mi = NRT_MemInfo_alloc(size);
+    return PyLong_FromVoidPtr(mi);
+}
+
+/*
+ * Like meminfo_alloc but set memory to zero after allocation and before
+ * deallocation.
+ */
+static
+PyObject*
+meminfo_alloc_safe(PyObject *self, PyObject *args) {
+    MemInfo *mi;
+    Py_ssize_t size;
+    if (!PyArg_ParseTuple(args, "K", &size)) {
+        return NULL;
+    }
+    mi = NRT_MemInfo_alloc_safe(size);
+    return PyLong_FromVoidPtr(mi);
+}
+
+typedef struct {
+    PyObject_HEAD
+    MemInfo *meminfo;
+    int      defer;
+} MemInfoObject;
+
+static
+int MemInfo_init(MemInfoObject *self, PyObject *args, PyObject *kwds) {
+    static char *keywords[] = {"ptr", NULL};
+    unsigned PY_LONG_LONG raw_ptr;
+    if (!PyArg_ParseTupleAndKeywords(args, kwds, "K", keywords, &raw_ptr)) {
+        return -1;
+    }
+    self->meminfo = (MemInfo*)raw_ptr;
+    self->defer = 0;
+    NRT_MemInfo_acquire(self->meminfo);
+    return 0;
+}
+
+int MemInfo_getbuffer(PyObject *exporter, Py_buffer *view, int flags) {
+    Py_ssize_t len;
+    void *buf;
+    int readonly = 0;
+
+    MemInfoObject *miobj = (MemInfoObject*)exporter;
+    MemInfo *mi = miobj->meminfo;
+
+    buf = NRT_MemInfo_data(mi);
+    len = NRT_MemInfo_size(mi);
+    return PyBuffer_FillInfo(view, exporter, buf, len, readonly, flags);
+}
+
+Py_ssize_t MemInfo_rdwrbufferproc(PyObject *self, Py_ssize_t segment,
+                                  void **ptrptr)
+{
+    MemInfoObject *mio = (MemInfoObject *)self;
+    MemInfo *mi = mio->meminfo;
+    if (segment != 0) {
+        PyErr_SetString(PyExc_TypeError, "MemInfo only has 1 segment");
+        return -1;
+    }
+    *ptrptr = NRT_MemInfo_data(mi);
+    return NRT_MemInfo_size(mi);
+}
+
+Py_ssize_t MemInfo_segcountproc(PyObject *self, Py_ssize_t *lenp) {
+    MemInfoObject *mio = (MemInfoObject *)self;
+    MemInfo *mi = mio->meminfo;
+    if (lenp) {
+        *lenp = NRT_MemInfo_size(mi);
+    }
+    return 1;
+}
+
+#if (PY_MAJOR_VERSION < 3)
+static PyBufferProcs MemInfo_bufferProcs = {MemInfo_rdwrbufferproc,
+                                            MemInfo_rdwrbufferproc,
+                                            MemInfo_segcountproc,
+                                            NULL};
+#else
+static PyBufferProcs MemInfo_bufferProcs = {MemInfo_getbuffer, NULL};
+#endif
+
+static
+PyObject*
+MemInfo_acquire(MemInfoObject *self) {
+    NRT_MemInfo_acquire(self->meminfo);
+    Py_RETURN_NONE;
+}
+
+static
+PyObject*
+MemInfo_release(MemInfoObject *self) {
+    NRT_MemInfo_release(self->meminfo, self->defer);
+    Py_RETURN_NONE;
+}
+
+static
+PyObject*
+MemInfo_release_defer(MemInfoObject *self) {
```

Comment:
Yes, I think getting rid of unused APIs will make the whole thing more palatable :-)